File: 02 백준/01 자료구조/Python/Prob10845.py

"""
title  : 10845. 큐 (Silver 4)
time   : 72ms       180ms
memory : 34028KB    115244KB
"""

import sys
from collections import deque

def solution():
    input = sys.stdin.readline

    q = deque()

    for _ in range(int(input())):
        command = list(input().rstrip().split())
        cm = command[0]

        if cm == 'push':
            q.append(int(command[1]))
        elif cm == 'pop':
            print(q.popleft() if q else -1)
        elif cm == 'size':
            print(len(q))
        elif cm == 'empty':
            print(0 if q else 1)
        elif cm == 'front':
            print(q[0] if q else -1)
        elif cm == 'back':
            print(q[-1] if q else -1)

solution()

File: 02 백준/01 자료구조/Java/src/Prob10845.java

/**
* title  : 10845. 큐 (Silver 4)
* time   : 168ms
* memory : 20584KB
*/

import java.io.*;
import java.util.*;

public class Prob10845 {
    
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(br.readLine());
        int lastElement = 0;

        Deque<Integer> q = new LinkedList<>();

        for (int i = 0; i < N; i++) {
            String[] comm = (br.readLine()).split(" ");

            switch(comm[0]) {

                case "push":
                    int num = Integer.parseInt(comm[1]);
                    q.offer(num);
                    lastElement = num;
                    break;

                case "pop":
                    sb.append(q.isEmpty() ? -1 : q.poll()).append("\n");
                    break;

                case "size":
                    sb.append(q.size()).append("\n");
                    break;

                case "empty":
                    sb.append(q.isEmpty() ? 1 : 0).append("\n");
                    break;

                case "front":
                    sb.append(q.isEmpty() ? -1 : q.peek()).append("\n");
                    break;

                case "back":
                    sb.append(q.isEmpty() ? -1 : lastElement).append("\n");
                    break;

                default:
                    break;
            }
        }
        System.out.println(sb);
    }
}
